### DISCUSSION TITLE
custom login logic(login with cookies, not using OAuth)

### DISCUSSION THREAD

**Discussion by canimu** [directus/directus#6989] [Archive]
Hey, there.
I need to customize the SSO function so that I can auto-login when I enter Directus admin app page
Since the client will send a request with the cookie, I can convert it to a username through my own library. I hope to use the prased username login to Directus without typing a password or using the OAuth function, and skip the Directus origin login page.

I've tried using API hooks to capture the login request, but the feature auth.refresh is still not released now. Hook login.* will be triggered when the login button had been pressed, seems not the solution too.
 
How can I skip the login page and automatically log in system with cookies? Can anybody help?

**Comment by rijkvanzanten** [comment]
> Since the client will send a request with the cookie, I can convert it to a username through my own library. I hope to use the prased username login to Directus without typing a password or using the OAuth function, and skip the Directus origin login page.

What client? Where is this cookie coming from?

**Reply by canimu** [reply]
The client is a browser, and the cookie is from another website with the same domain.

The full process is like below:
1.Our user login at one website with the domain "mycompany.com"
2.After successfully logged in, the user now has a token stored in their cookie "MY_LOGIN_TOKEN" with domain "mycompany.com"
3.And then the user will browse to Directus which we will host on the same domain, ex: "directus.mycompany.com", so the cookie "MY_LOGIN_TOKEN" will be sent to our Directus backend. I want to get this cookie and resolve it to username and automatically login with the same username stored in Directus User Database.

**Reply by rijkvanzanten** [reply]
I don't think there's a way to do that currently

**Reply by canimu** [reply]
I got it. Thanks for your suggestion!

**Reply by Abbas-Abdelila** [reply]
Is there a way to do it now currently? I am also looking for a one click login from dashboard.mywebsite.com that has all the user information including password to directus instance without seeing the login to directus CMS dashboard.
